wallet/chain/p/builder/builder_test.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package builder
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils"
+	"github.com/ava-labs/avalanchego/vms/components/avax"
+	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/platformvm/stakeable"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
+)
+
+//nolint:gosec // This does not need cryptographically strong random numbers
+func generateUTXOs(assetID ids.ID, locktime uint64) []*avax.UTXO {
+	utxos := make([]*avax.UTXO, rand.Intn(10))
+	for i := range utxos {
+		var output avax.TransferableOut = &secp256k1fx.TransferOutput{
+			Amt: rand.Uint64(),
+			OutputOwners: secp256k1fx.OutputOwners{
+				Locktime:  rand.Uint64(),
+				Threshold: 1,
+				Addrs:     []ids.ShortID{ids.GenerateTestShortID()},
+			},
+		}
+		if locktime != 0 {
+			output = &stakeable.LockOut{
+				Locktime:        locktime,
+				TransferableOut: output,
+			}
+		}
+		utxos[i] = &avax.UTXO{
+			UTXOID: avax.UTXOID{
+				TxID:        ids.GenerateTestID(),
+				OutputIndex: rand.Uint32(),
+			},
+			Asset: avax.Asset{
+				ID: assetID,
+			},
+			Out: output,
+		}
+	}
+	return utxos
+}
+
+func TestSplitUTXOsByLocktime(t *testing.T) {
+	var (
+		require = require.New(t)
+
+		expectedUnlockedUTXOsPart0 = generateUTXOs(ids.GenerateTestID(), 0)
```

Comment:
I added a variable to hopefully make the code more clear here

Reply:
`unlockedTime`

`OutputOwners.Locktime` is when the UTXO can be spent.
`Lockout.Locktime` is when the UTXO can be transferred to a new owner.

So, if `OutputOwners.Locktime <= now < Lockout.Locktime`, then the UTXO can be spent. However, it can not be transferred to a new owner.